spec/models/manageiq/providers/openshift/container_manager/refresher_spec.rb

```diff
@@ -127,6 +127,115 @@ def normal_refresh
     assert_specific_unused_container_image(:metadata => true, :connected => false)
   end
 
+  context "when refreshing an empty DB" do
+    # CREATING FIRST VCR
+    # To recreate the tested objects in OpenShift use the template file:
+    # spec/vcr_cassettes/manageiq/providers/openshift/container_manager/test_objects_template.yml
+    # and the following commands for 3 projects my-project-X (X=0/1/2):
+    # oc new-project my-project-X
+    # oc process -f template.yml -v INDEX=X | oc create -f -
+
+    before(:each) do
+      VCR.use_cassette("#{described_class.name.underscore}_before_openshift_deletions",
+                       :match_requests_on => [:path,]) do # , :record => :new_episodes) do
+        EmsRefresh.refresh(@ems)
+      end
+    end
+
+    it "saves the objects in the DB" do
+      expect(ContainerProject.count).to eq(8)
+      expect(ContainerImage.count).to eq(44)
+      expect(ContainerRoute.count).to eq(6)
+      expect(ContainerTemplate.count).to eq(30)
+      expect(ContainerReplicator.count).to eq(10)
+      expect(ContainerBuild.count).to eq(3)
+      expect(ContainerBuildPod.count).to eq(3)
+      expect(CustomAttribute.count).to eq(532)
+      expect(ContainerTemplateParameter.count).to eq(264)
+      expect(ContainerRoute.find_by(:name => "my-route-2").labels.count).to eq(1)
+      expect(ContainerTemplate.find_by(:name => "my-template-2").container_template_parameters.count).to eq(1)
+    end
+
+    context "when refreshing non empty DB" do
+      # CREATING SECOND VCR
+      # To delete the tested objects in OpenShift use the following commands:
+      # oc delete project my-project-0
+      # oc project my-project-1
+      # oc delete pod my-pod-1
+      # oc delete service my-service-1
+      # oc delete route my-route-1
+      # oc delete resourceQuota my-resource-quota-1
+      # oc delete limitRange my-limit-range-1
+      # oc delete persistentVolumeClaim my-persistentvolumeclaim-1
+      # oc delete template my-template-1
+      # oc delete build my-build-1
+      # oc delete buildconfig my-build-config-1
+      # oc project my-project-2
+      # oc label route my-route-2 key-route-label-
+      # oc edit template my-template-2 # remove the template parameters from the file and save it
+      # oc delete pod my-pod-2
+
+      before(:each) do
+        VCR.use_cassette("#{described_class.name.underscore}_after_openshift_deletions",
+                         :match_requests_on => [:path,]) do # , :record => :new_episodes) do
+          EmsRefresh.refresh(@ems)
+        end
+      end
+
+      it "archives objects" do
+        expect(ContainerProject.count).to eq(8)
+        expect(ContainerProject.where(:deleted_on => nil).count).to eq(7)
+        expect(ContainerImage.count).to eq(43) # should be 44
+        expect(ContainerImage.where(:deleted_on => nil).count).to eq(43) # should be 44
```

Comment:
yeah I mentioned it in a comment before but it is hidden writing it again: 

I don't know what is the expected behavior here.
We save "2 kinds" of images in the DB:

Images that were imported into the OpenShfit cluster which we get as all other resources in the inventory.
Images that were not imported to the cluster, but are in use by one or more containers.
If an image of the first kind is deleted in OpenShift we disconnect it on our side (do not delete it). However, deletion of all containers that run an image of the second kind causes a deletion of that image from the DB (not disconnect).
cc @simon3z @enoodle